modules/ballerina-core/src/main/java/org/ballerinalang/runtime/ServerConnectorMessageHandler.java

```diff
@@ -150,8 +151,13 @@ public static void  invokeResource(CarbonMessage cMsg, CarbonCallback callback,
         Context context = new Context(programFile);
         context.setServiceInfo(serviceInfo);
         context.setCarbonMessage(resourceMessage);
-        context.setProperty("SRC_HANDLER", resourceMessage.getProperty("SRC_HANDLER"));
         context.setBalCallback(new DefaultBalCallback(resourceCallback));
+
+        SourceHandler srcHandler = null;
+        if (resourceMessage.getProperty(Constants.SRC_HANDLER) != null) {
+            srcHandler = (SourceHandler) resourceMessage.getProperty(Constants.SRC_HANDLER);
+            context.setProperty(Constants.SRC_HANDLER, srcHandler);
+        }
```

Comment:
I think what Shafreen means is line number 144 is not needed, but line number 146 is needed.